### PR TITLE
Add new checkly_env_var resource to support Checkly environment variables

### LINKED ISSUE #41: Environment Variables Resource
Add resource for Environment Variables

### PR DESCRIPTION
**Pull request by ianaya89**
## Affected Components
* [x] Resources
* [x] Tests
* [x] Docs
* [ ] Other

## Style
* [x] Terraform code is formatted with `terraform fmt`
* [x] Go code is formatted with `go fmt`

## Notes for the Reviewer
Add new `checkly_env_var` resource to support Checkly environment variables

> Resolves #41

### CODE REVIEW

File: checkly/resource_env_var.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+func resourceEnvVar() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceEnvVarCreate,
+		Read:   resourceEnvVarRead,
+		Update: resourceEnvVarUpdate,
+		Delete: resourceEnvVarDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
```
Comment (tnolet) [inline checkly/resource_env_var.go:22]: This schema is missing the `locked` property which we have on the API. This is a simple boolean.

**Review by tnolet — COMMENTED** [review]
I think we want to address our environment variables API endpoint first. We have the issue that we do not have a specific ID for a variable. The `key` can work, but we now allow multiple `keys` with the same value. On a `GET` you just get the last one you added.

**Comment by ianaya89** [comment]
@tnolet makes sense. This PR will only valid if we assume env `key` as the unique identifier. In that case, we should put this on hold until we add the backend restriction

**Review by pilimartinez — APPROVED** [review]
👍

**Pull request merged**